evalml/tests/conftest.py

```diff
@@ -439,6 +539,15 @@ class NonLinearRegressionPipeline(RegressionPipeline):
             'Elastic Net': ['Elastic Net Regressor', 'OneHot.x'],
             'Linear Regressor': ['Linear Regressor', 'Random Forest', 'Elastic Net']
         }
+
+        def __init__(self, parameters, random_seed=0):
+            return super().__init__(self.component_graph, None, parameters)
+
+        def new(self, parameters, random_seed=0):
+            return self.__class__(parameters, random_seed)
+
+        def clone(self):
+            return self.__class__(self.parameters, random_seed=self.random_seed)
     return NonLinearRegressionPipeline
```

Comment:
I know I'm guilty of this, but do you think we need some concerted effort to move the testing pipelines that are generated in other test files etc here to conftest?  It's not at all in the scope of this PR, I would just wonder what your recommendations are, having gone through all this, to make the next sweeping change easier?

Reply:
I think for most cases, pipelines that are shared amongst files are stored here, but we also create a lot of classes within tests to test for specific things. Some of these can be cleaned up, others are specific (ex: testing Catboost). Hopefully, this redesign means that we don't need to follow this pattern of creating classes for tests. I've left the ones in conftest alone because I figured in cases of reusability, maybe having a class is actually the way to go. In this PR I've also tried to rip out those one-off cases where we define a class for a specific test :')